### DISCUSSION TITLE
Run ftp upload from two ues to different DNs  at the sametime

### DISCUSSION THREAD

**Discussion by liuywa** [aligungr/UERANSIM#484] [Q&A]
When i run ftp upload from two ues to different DNs at the sametime, i will got the bellow errors from nr-ue and some warnings from nr-gnb after runing a while. Could you tell is it an known issue or what did i do wrong? Thanks!

**command i used is:**
> ./nr-gnb -c my-gnb.yaml
> ./nr-ue -c my-ue.yaml -n 2

**error from nr-ue:**
> [2022-02-15 00:26:11.116] [286010000000002|app] [info] Connection setup for PDU session[1] is successful, TUN interface[uesimtun0, 10.10.5.12] is up.
> [2022-02-15 00:26:11.165] [286010000000001|app] [info] Connection setup for PDU session[1] is successful, TUN interface[uesimtun1, 10.10.5.11] is up.
> [2022-02-15 00:32:18.892] [286010000000002|rrc] [debug] Signal lost for cell[1], total [0] cells in coverage
> [2022-02-15 00:32:18.893] [286010000000002|rrc] [debug] New signal detected for cell[2], total [1] cells in coverage
> [2022-02-15 00:32:18.893] [286010000000002|nas] [error] Radio link failure detected
> [2022-02-15 00:32:18.893] [286010000000002|nas] [info] UE switches to state [CM-IDLE]
> [2022-02-15 00:32:18.894] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/PS]
> [2022-02-15 00:32:18.894] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/PLMN-SEARCH]
> [2022-02-15 00:32:18.894] [286010000000002|nas] [error] PLMN selection failure, no cells in coverage
> [2022-02-15 00:32:20.995] [286010000000002|rrc] [warning] Acceptable cell selection failed in [1] cells. [1] no SI, [0] reserved, [0] barred, ftai [0]
> [2022-02-15 00:32:20.995] [286010000000002|rrc] [error] Cell selection failure, no suitable or acceptable cell found
> [2022-02-15 00:32:21.896] [286010000000002|nas] [error] PLMN selection failure, no cells in coverage
> [2022-02-15 00:32:22.231] [286010000000002|nas] [info] Selected plmn[286/01]
> [2022-02-15 00:32:22.231] [286010000000002|rrc] [info] Selected cell plmn[286/01] tac[1] category[SUITABLE]
> [2022-02-15 00:32:22.231] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/PS]
> [2022-02-15 00:32:22.232] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/NORMAL-SERVICE]
> [2022-02-15 00:32:25.202] [286010000000002|nas] [debug] Uplink data status changed PSI[1] pending[true]
> [2022-02-15 00:32:25.202] [286010000000002|nas] [debug] Service request required due to [IDLE-UPLINK-DATA-PENDING]
> [2022-02-15 00:32:25.202] [286010000000002|nas] [debug] UAC access attempt is allowed for identity[1, 2], category[MO_data]
> [2022-02-15 00:32:25.202] [286010000000002|nas] [debug] Sending Service Request due to [IDLE-UPLINK-DATA-PENDING]
> [2022-02-15 00:32:25.204] [286010000000002|nas] [info] UE switches to state [MM-SERVICE-REQUEST-INITIATED]
> [2022-02-15 00:32:25.204] [286010000000002|rrc] [debug] Sending RRC Setup Request
> [2022-02-15 00:32:40.253] [286010000000002|nas] [debug] NAS timer[3517] expired [1]
> [2022-02-15 00:32:40.254] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/PS]
> [2022-02-15 00:32:40.254] [286010000000002|nas] [info] UE switches to state [MM-REGISTERED/NORMAL-SERVICE]
> [2022-02-15 00:37:19.532] [286010000000002|nas] [debug] NAS timer[3512] expired [1]
> [2022-02-15 00:37:19.533] [286010000000002|nas] [debug] Mobility registration updating required due to [T3512-EXPIRY]
> [2022-02-15 00:37:19.534] [286010000000002|nas] [debug] UAC access attempt is allowed for identity[1, 2], category[MO_data]
> [2022-02-15 00:37:19.534] [286010000000002|nas] [debug] Sending Periodic Registration with update cause [T3512-EXPIRY]
> [2022-02-15 00:37:19.537] [286010000000002|nas] [info] UE switches to state [MM-REGISTER-INITIATED]
> [2022-02-15 00:37:19.537] [286010000000002|rrc] [debug] Sending RRC Setup Request
> [2022-02-15 00:37:34.547] [286010000000002|nas] [debug] NAS timer[3510] expired [1]

**warning from nr-gnb:**
> UERANSIM v3.2.6
> [2022-02-15 00:25:42.184] [sctp] [info] Trying to establish SCTP connection... (10.10.8.10:38412)
> [2022-02-15 00:25:42.254] [sctp] [info] SCTP connection established (10.10.8.10:38412)
> [2022-02-15 00:25:42.255] [sctp] [debug] SCTP association setup ascId[1]
> [2022-02-15 00:25:42.256] [ngap] [debug] Sending NG Setup Request
> [2022-02-15 00:25:42.325] [ngap] [debug] NG Setup Response received
> [2022-02-15 00:25:42.325] [ngap] [info] NG Setup procedure is successful
> [2022-02-15 00:26:10.849] [rrc] [debug] UE[1] new signal detected
> [2022-02-15 00:26:10.850] [rrc] [debug] UE[2] new signal detected
> [2022-02-15 00:26:10.853] [rrc] [info] RRC Setup for UE[1]
> [2022-02-15 00:26:10.853] [rrc] [info] RRC Setup for UE[2]
> [2022-02-15 00:26:10.854] [ngap] [debug] Initial NAS message received from UE[1]
> [2022-02-15 00:26:10.854] [ngap] [debug] Initial NAS message received from UE[2]
> [2022-02-15 00:26:10.880] [ngap] [debug] Initial Context Setup Request received
> [2022-02-15 00:26:10.884] [ngap] [debug] Initial Context Setup Request received
> [2022-02-15 00:26:10.948] [ngap] [info] PDU session resource(s) setup for UE[1] count[1]
> [2022-02-15 00:26:10.995] [ngap] [info] PDU session resource(s) setup for UE[2] count[1]
> [2022-02-15 00:32:25.206] [rrc] [warning] Discarding RRC Setup Request, UE context already exists
> [2022-02-15 00:37:19.538] [rrc] [warning] Discarding RRC Setup Request, UE context already exists

**Comment by aligungr** [comment]
Hello @liuywa

Could you describe how to reproduce this error? so that I can investigate it quickly.

**Reply by liuywa** [reply]
I am sorry for getting back to you so late!
First you should create two vitual machines used for DN1 and DN2 and deploy ftp server on them.
Then could run nr-gnb and nr-ue on on your environment. After the ues access and pdu session create successful, you could execute the bellow command to do ftp upload. You can see the attach file for details! Thanks!
ftp 12.12.12.1 //logon the DN1 ftp server
ftp>put xxx.  //upload xxx file

ftp 12.12.13.1 //logon the DN2 ftp server
ftp>put xxx //upload xxx file
![nr-gnb-test](https://user-images.githubusercontent.com/16411376/155301724-70050851-bfe9-458e-b69b-cf456c2ad4ad.png)
